libs/wingsdk/src/std/json.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * Immutable Json
+ */
+export class Json {
+  /**
+   * Returns a specified element from the Json.
+   *
+   * @macro ($self$)[$args$]
+   *
+   * @param key The key of the element to return
+   * @returns The element associated with the specified key, or undefined if the key can't be found
+   */
+  public get(key: string): Json {
+    key;
+    throw new Error("Macro");
+  }
+
+  /**
+   * Returns a specified element at a given index from Json Array
+   *
+   * @macro ($self$)[$args$]
+   *
+   * @param index The index of the element in the Json Array to return
+   * @returns The element at given index in Json Array, or undefined if index is not valid
+   */
+  public getAt(index: number): Json {
+    index;
+    throw new Error("Macro");
+  }
+}
+
+/**
+ * Mutable Json
+ */
+export class MutJson {
```

Comment:
Can we have `MutJson` extend `Json` to avoid duplicated methods? We did the same at array.ts.
```suggestion
export class MutJson extends Json {
```

Reply:
The issue is that Json get methods return Json/MutJson whereas the arrays return a generic type. So for instance if MutJson class extended Json and did not override the get method it would return a `Json` when using .get() which means you couldn't do something like 

```ts
let x = MutJson {
	y: {
		z: "cool"
	}
}

let a = x.get("y");

a.set("z", "beans");
   ^^^ No set method on Json
```

This could be solved potentially with a behavioral interface, but I think we might rethink this entirely with `dot` and `[]` notation.

Reply:
Ok, thanks for clarifying. I guess we could have `get` return the generic type T1, no?
In any case, this might suffice for now as we'll give it another pass when we implement the `dot` notation.